include/llvm-dialects/Dialect/Dialect.h

```diff
@@ -245,11 +245,14 @@ class DialectExtensionRegistration {
 
 namespace detail {
 
-bool isSimpleOperationDecl(const llvm::Function *fn, llvm::StringRef name);
-bool isOverloadedOperationDecl(const llvm::Function *fn, llvm::StringRef name);
-
-bool isSimpleOperation(const llvm::CallInst *i, llvm::StringRef name);
-bool isOverloadedOperation(const llvm::CallInst *i, llvm::StringRef name);
+bool isSimpleOperationDecl(const llvm::Function *fn, llvm::StringRef mnemonic);
+bool isOverloadedOperationDecl(const llvm::Function *fn,
+                               llvm::StringRef mnemonic);
+
+bool isSimpleOperation(const llvm::CallInst *i, llvm::StringRef mnemonic);
+bool isOverloadedOperation(const llvm::CallInst *i, llvm::StringRef mnemonic);
+bool isOperationDecl(llvm::StringRef fn, bool isOverloaded,
+                     llvm::StringRef mnemonic);
```

Comment:
I still think this would be better served with a Function argument for consistency. I think the one place it's used without a function in OpSet could be solved like in OpMap instead.